payjoin-defaults/src/lib.rs

```diff
@@ -0,0 +1,149 @@
+use payjoin::Url;
+
+/// Fetch the ohttp keys from the specified payjoin directory via proxy.
+///
+/// * `ohttp_relay`: The http CONNNECT method proxy to request the ohttp keys from a payjoin
+/// directory.  Proxying requests for ohttp keys ensures a client IP address is never revealed to
+/// the payjoin directory.
+///
+/// * `payjoin_directory`: The payjoin directory from which to fetch the ohttp keys.  This
+/// directory stores and forwards payjoin client payloads.
+///
+/// * `cert_der` (optional): The DER-encoded certificate to use for local HTTPS connections.  This
+/// parameter is only available when the "danger-local-https" feature is enabled.
+#[cfg(feature = "v2")]
+pub fn fetch_ohttp_keys(
+    ohttp_relay: Url,
+    payjoin_directory: Url,
+    #[cfg(feature = "danger-local-https")] cert_der: Vec<u8>,
+) -> Result<payjoin::OhttpKeys, Error> {
+    let ohttp_keys_url = payjoin_directory.join("/ohttp-keys")?;
+    let proxy = PayjoinProxy::new(
+        &ohttp_relay,
+        #[cfg(feature = "danger-local-https")]
+        cert_der,
+    )?;
+    let res = proxy.get(ohttp_keys_url.as_str()).call()?;
```

Comment:
Since proxy is your own abstraction, it probably makes sense more to have it take your abstraction (Url). Otherwise, what's the point of the abstraction? I don't really think we need this abstraction / impl layer though so I'm OK ignoring it for now.